MIG-1509: MTC 1.7.15 Release Notes (#75137)
openshift/openshift-docs: modules/migration-mtc-release-notes-1-7-15.adoc

== Resolved issues

This release has the following resolved issues:

.CVE-2024-24786: A flaw was found in Golang's protobuf module, where the unmarshal function can enter an infinite loop

A flaw was found in the `protojson.Unmarshal` function, which could enter an infinite loop when unmarshaling certain forms of invalid JSON. This condition could occur when unmarshaling into a message which contained a `google.protobuf.Any` value, or when the `UnmarshalOptions.DiscardUnknown` option was set.

To resolve this issue, upgrade to {mtc-short} 1.7.15.

For more details, see link:https://access.redhat.com/security/cve/CVE-2024-24786[(CVE-2024-24786)]

.CVE-2024-28180: `jose-go` improper handling of highly compressed data

A vulnerability was found in Jose due to improper handling of highly compressed data. An attacker could send a JSON Web Encryption (JWE) containing compressed data that used large amounts of memory and CPU when decompressed by Decrypt or DecryptMulti. 

To resolve this issue, upgrade to {mtc-short} 1.7.15.

For more details, see link:https://access.redhat.com/security/cve/CVE-2024-28180[(CVE-2024-28180)]

[id="known-issues-1-7-15_{context}"]
== Known issues

> There are no major known issues in this release.

Author: ```suggestion
This release has the following known issues:
```

// Module included in the following assemblies:
//
// * migration_toolkit_for_containers/mtc-release-notes.adoc
:_mod-docs-content-type: REFERENCE
[id="migration-mtc-release-notes-1-7-15_{context}"]
= {mtc-full} 1.7.15 release notes

[id="resolved-issues-1-7-15_{context}"]
== Resolved issues

This release has the following resolved issues:

.CVE-2024-24786: A flaw was found in Golang's protobuf module, where the unmarshal function can enter an infinite loop

> A flaw was found in the `protojson.Unmarshal` function, which could enter an infinite loop when unmarshaling certain forms of invalid JSON. This condition could occur when unmarshaling into a message that contained a `google.protobuf.Any` value or when the `UnmarshalOptions.DiscardUnknown` option was set.

Reviewer: "which" is typically used for non-restrictive clauses but this statement seems essential· How about:

> A flaw was found in the `protojson.Unmarshal` function that could cause the function to enter an infinite loop when unmarshaling certain forms of invalid JSON messages. This condition could occur when unmarshaling into a message that contained a `google.protobuf.Any` value or when the `UnmarshalOptions.DiscardUnknown` option was set in a JSON-formatted message.

> A flaw was found in the `protojson.Unmarshal` function, which could enter an infinite loop when unmarshaling certain forms of invalid JSON. This condition could occur when unmarshaling into a message that contained a `google.protobuf.Any` value or when the `UnmarshalOptions.DiscardUnknown` option was set.

Author: ```suggestion
A flaw was found in the `protojson.Unmarshal` function that could cause the function to enter an infinite loop when unmarshaling certain forms of invalid JSON messages. This condition could occur when unmarshaling into a message that contained a `google.protobuf.Any` value or when the `UnmarshalOptions.DiscardUnknown` option was set in a JSON-formatted message.
```

To resolve this issue, upgrade to {mtc-short} 1.7.15.

For more details, see link:https://access.redhat.com/security/cve/CVE-2024-24786[(CVE-2024-24786)].

.CVE-2024-28180: `jose-go` improper handling of highly compressed data

> A vulnerability was found in Jose due to improper handling of highly compressed data. An attacker could send a JSON Web Encryption (JWE) containing compressed data that used large amounts of memory and CPU when decompressed by Decrypt or DecryptMulti. 

Reviewer: ```suggestion
A vulnerability was found in Jose due to improper handling of highly compressed data. An attacker could send a JSON Web Encryption (JWE) encrypted message that contained compressed data that used large amounts of memory and CPU when decompressed by `Decrypt` or `DecryptMulti` vector/function. 
```

Reviewer: Vector or function?

To resolve this issue, upgrade to {mtc-short} 1.7.15.

For more details, see link:https://access.redhat.com/security/cve/CVE-2024-28180[(CVE-2024-28180)].

[id="known-issues-1-7-15_{context}"]
== Known issues

This release has the following known issues:

> .Direct Volume Migration is failing as the rsync pod on the source cluster goes into an `Error` state

Reviewer: ```suggestion
.Direct Volume Migration is failing as the Rsync pod on the source cluster goes into an `Error` state
```

Reviewer: As per https://docs.openshift.com/container-platform/4.15/migration_toolkit_for_containers/advanced-migration-options-mtc.html ?

> On migrating any application with Persistent Volume Claim (PVC), Stage succeeds with warnings and Direct Volume Migration (DVM) fails with the `rsync` pod on the source namespace going into an `error` state. link:https://bugzilla.redhat.com/show_bug.cgi?id=2256141[(BZ#2256141)]

Reviewer: "On migrating any application with Persistent Volume Claim (PVC)"

Not sure what is meant here:

> An application that contains a PVC or migrating an application by using a PVC.

Reviewer: "Stage" is a migration operation? Might be worth stating this so a supporting noun is evident.

.The conflict condition is briefly cleared after it is created

> When creating a new state migration plan that will result in a conflict error, that error is cleared shortly after it is displayed. link:https://bugzilla.redhat.com/show_bug.cgi?id=2144299[(BZ#2144299)]

Reviewer: ```suggestion
When creating a new state migration plan that results in a conflict error message, the error message is cleared shortly after it is displayed. link:https://bugzilla.redhat.com/show_bug.cgi?id=2144299[(BZ#2144299)]
```

Displayed on a terminal?

> .Migration fails when there are multiple Volume Snapshot Locations (VSLs) of different provider types configured in a cluster and no default VSL is set

Reviewer: ```suggestion
.Migration fails when there are multiple Volume Snapshot Locations (VSLs) of different provider types configured in a cluster with no specified default VSL.
```

> When there are multiple VSLs in a cluster with different provider types, and none of them is set as default, Velero results in a validation error, making migrations fail.  link:https://bugzilla.redhat.com/show_bug.cgi?id=2180565[(BZ#2180565)]

Reviewer: ```suggestion
When there are multiple VSLs in a cluster with different provider types, and you set none of them  as the default VSL, Velero results in a validation error that cuases migration operations to fail.  link:https://bugzilla.redhat.com/show_bug.cgi?id=2180565[(BZ#2180565)]
```

> When there are multiple VSLs in a cluster with different provider types, and none of them is set as default, Velero results in a validation error, making migrations fail.  link:https://bugzilla.redhat.com/show_bug.cgi?id=2180565[(BZ#2180565)]

Author: ```suggestion
When there are multiple VSLs in a cluster with different provider types, and you set none of them as the default VSL, Velero results in a validation error that causes migration operations to fail. link:https://bugzilla.redhat.com/show_bug.cgi?id=2180565[(BZ#2180565)]
```
